sklearn/utils/_estimator_html_repr.py

```diff
@@ -381,10 +261,36 @@ def estimator_html_repr(estimator):
     html: str
         HTML representation of estimator.
     """
+    from sklearn.exceptions import NotFittedError
+    from sklearn.utils.validation import check_is_fitted
+
+    if not hasattr(estimator, "fit"):
```

Comment:
What do you think about having a third visual state to represent estimators that can't be fitted? It seems odd to use "not fitted" as the state for an estimator that can't be fitted. Especially if you think of red/orange kinds of colours as "warning" or "danger". After all there is nothing at all wrong with this kind of estimator, the only thing that is special is that it is stateless. If you display a whole pipeline and have a stateless one in there, you get all blue except for one orange :-/ Maybe the right colour for a stateless estimator is the same colour as for a fitted estimator? They are both "ready to be used".

WDYT?

Reply:
This is a good point, and definitely something that should be addressed. For this PR, however, we thought it would be better to get the simple case working and implement the fitted/unfitted cases first, then address the more complex cases (@glemaitre mentioned frozen estimators) later. 

I would say it's a great suggestion for a later PR, maybe?

Reply:
As it is right now, the check for `fitted` is done in the first call of `estimator_html_repr`, which means that if the "highest level estimator" (e.g., `Pipeline`) is fitted, then everything else is treated as fitted. (this was simpler to implement) 

To properly implement your suggestion, the entire block checking for whether the estimator is fitted or not should be moved inside `_write_estimator_html`. If that's done, then we could in principle add as many states as we want (provided the css file is updated accordingly). 

I don't think it's too complex to do, but I still think this should be done in a separate PR, also due to the fact we need the tests for it. 

@GaelVaroquaux what do you think?

Reply:
Ok. So the result of "highest level estimator is checked"  would be that if you have a stateless estimator in your pipeline then it will be displayed as "fitted", if the pipeline itself has been fitted. This means the weird case I was thinking of, where you have every step of your pipeline being displayed as fitted except the stateless ones doesn't occur.

If that is correct, then I think we should indeed punt this to a new PR.